tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_scalar_comparison.rs

```diff
@@ -398,6 +398,41 @@ fn integer_unchecked_scalar_comparisons_edge(param: ClassicPBSParameters) {
     }
 }
 
+// Tests whether equality holds for greater/less or equal than, but fails for greater/lesser than.
+fn integer_comparisons_equality_on_edge(param: ClassicPBSParameters) {
+    let num_blocks = (128f64 / (param.message_modulus.0 as f64).log(2.0)).ceil() as usize;
+    let (cks, sks) = KEY_CACHE.get_from_params(param, IntegerKeyKind::Radix);
+
+    let mut rng = rand::thread_rng();
+
+    let scalar = rng.gen::<u64>();
+    let ct = cks.encrypt_radix(scalar, num_blocks);
+
+    {
+        let result = sks.unchecked_scalar_ge_parallelized(&ct, scalar);
+        let decrypted = cks.decrypt_bool(&result);
+        assert_eq!(decrypted, true);
+    }
+
+    {
+        let result = sks.unchecked_scalar_le_parallelized(&ct, scalar);
+        let decrypted = cks.decrypt_bool(&result);
+        assert_eq!(decrypted, true);
+    }
+
+    {
+        let result = sks.unchecked_scalar_gt_parallelized(&ct, scalar);
+        let decrypted = cks.decrypt_bool(&result);
+        assert_eq!(decrypted, false);
+    }
+
+    {
+        let result = sks.unchecked_scalar_lt_parallelized(&ct, scalar);
+        let decrypted = cks.decrypt_bool(&result);
+        assert_eq!(decrypted, false);
+    }
+}
+
```

Comment:
That test is not useful I believe, as we already test these cases, and more importantly its won't trigger the case you are fixing

The fix you made is for when the encrypted radix is empty (no blocks)

Reply:
Thanks for looking into this! I addressed both of your comments.